subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/NativePlatformBackedFileMetadataAccessor.java

```diff
@@ -35,21 +35,25 @@ public NativePlatformBackedFileMetadataAccessor(Files files) {
 
     @Override
     public FileMetadataSnapshot stat(File f) {
-        FileInfo stat = files.stat(f, true);
-        switch (stat.getType()) {
-            case File:
-                return DefaultFileMetadataSnapshot.file(stat.getLastModifiedTime(), stat.getSize());
-            case Directory:
-                return DefaultFileMetadataSnapshot.directory();
-            case Missing:
-                return DefaultFileMetadataSnapshot.missing();
-            default:
-                throw new IllegalArgumentException("Unrecognised file type: " + stat.getType());
+        try {
+            FileInfo stat = files.stat(f, true);
+            switch (stat.getType()) {
+                case File:
+                    return DefaultFileMetadataSnapshot.file(stat.getLastModifiedTime(), stat.getSize());
+                case Directory:
+                    return DefaultFileMetadataSnapshot.directory();
+                case Missing:
+                    return DefaultFileMetadataSnapshot.missing();
+                default:
+                    throw new IllegalArgumentException("Unrecognised file type: " + stat.getType());
+            }
+        } catch (NativeException e) {
+            return DefaultFileMetadataSnapshot.missing();
         }
```

Comment:
We should wrap the `stat` calls only.

Reply:
Also, can we somehow be more specific about what we catch here? I don't like the fact that we catch a generic `NativeException`.

Reply:
We do the same for Java NIO, and the old Java API doesn't throw exceptions at all. I would catch `NativeException` here, since that is our current logic.